pkg/controller/replicaset/replica_set.go

```diff
@@ -606,18 +595,8 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 			filteredPods = append(filteredPods, pod)
 		}
 	}
-	// If any adoptions are attempted, we should first recheck for deletion with
-	// an uncached quorum read sometime after listing Pods (see #42639).
-	canAdoptFunc := controller.RecheckDeletionTimestamp(func() (metav1.Object, error) {
-		fresh, err := rsc.kubeClient.ExtensionsV1beta1().ReplicaSets(rs.Namespace).Get(rs.Name, metav1.GetOptions{})
-		if err != nil {
-			return nil, err
-		}
-		if fresh.UID != rs.UID {
-			return nil, fmt.Errorf("original ReplicaSet %v/%v is gone: got uid %v, wanted %v", rs.Namespace, rs.Name, fresh.UID, rs.UID)
-		}
-		return fresh, nil
-	})
+
+	canAdoptFunc := getCanAdoptFunc(rsc, rs)
```

Comment:
On second thought, I'm thinking we should keep this as is, but move everything from here down to `cm.ClaimPods` into a method `rsc.claimPods()`. That makes more sense to me as a unit.

Reply:
Yes, this change will be better. Should we add unit tests for the `rsc.claimPods()`?